src/workerd/api/cache.c++

```diff
@@ -44,13 +44,15 @@ kj::StringPtr validateUrl(kj::StringPtr url) {
   //   But, that might mean e.g. discarding fragments ("hashes", stuff after a '#'), which would
   //   be a change in behavior that could subtly affect production workers...
 
-  constexpr auto urlOptions = kj::Url::Options { .percentDecode = false, .allowEmpty = true };
-  KJ_IF_MAYBE(parsed, kj::Url::tryParse(url, kj::Url::HTTP_PROXY_REQUEST, urlOptions)) {
-    return url;
-  } else {
-    JSG_FAIL_REQUIRE(TypeError,
-                      "Invalid URL. Cache API keys must be fully-qualified, valid URLs.");
-  }
+  static constexpr auto urlOptions = kj::Url::Options {
+    .percentDecode = false,
+    .allowEmpty = true,
+  };
+
+  JSG_REQUIRE(kj::Url::tryParse(url, kj::Url::HTTP_PROXY_REQUEST, urlOptions) != kj::none,
+              TypeError, "Invalid URL. Cache API keys must be fully-qualified, valid URLs.");
+
+  return url;
```

Comment:
hmm, maybe this isn't feasible or at least not trivial, but while we're changing kj::Maybe semantics, I wonder if we could also make the `kj::Maybe<T>` to `T` conversion explicit? Maybe this was already discussed and I missed it, if so apologies.

CC @harrishancock